lib/spack/spack/environment/environment.py

```diff
@@ -599,6 +689,16 @@ def regenerate(self, concretized_root_specs):
             tty.debug("View at %s does not need regeneration." % self.root)
             return
 
+        if update_method == "exchange" and os.path.isdir(new_root):
```

Comment:
Just to double-check, do you need to avoid following symlinks here (both isdir and stat are used).

Reply:
No, that's already been checked if you're using the "exchange" method